rtos/MemoryPool.h

```diff
@@ -83,7 +84,10 @@ class MemoryPool : private mbed::NonCopyable<MemoryPool<T, pool_sz> > {
 
     /** Free a memory block.
       @param   block  address of the allocated memory block to be freed.
-      @return         status code that indicates the execution status of the function.
+      @return         osOK on successful deallocation, osErrorParameter if parameter mp_id
```

Comment:
what is mb_id in our case? From the perspective of C++ API it doesn't exist. I don't think we should be quoting it, instead we could say something like internal error for both.